web-api/src/personalization/routes.rs

```diff
@@ -116,12 +120,17 @@ async fn update_interactions(
 
 /// Represents personalized documents query params.
 #[derive(Debug, Clone, Deserialize)]
-struct PersonalizedDocumentsQuery {
-    count: Option<usize>,
+pub(crate) struct PersonalizedDocumentsQuery {
+    pub(crate) count: Option<usize>,
+    /// Personalize only the specified documents, otherwise fetch documents to be personalized.
+    // Note: documents can't be provided via the public server api, since Query<Self> always fails
+    // to deserialize if documents is present in a query string (because it's a sequence) and the
+    // route returns a bad request
+    pub(crate) documents: Option<Vec<DocumentId>>,
```

Comment:
:wrench: We should touch the production endpoint only to refactor the code and nothing else. The property for which this is not exposed to the public API is not trivial and we should keep it simple.

The functionalities for the tests can be implemented in dedicated handlers that have code in common with this one.